source/fuzz/transformation_load.cpp

```diff
@@ -61,15 +67,70 @@ bool TransformationLoad::IsApplicable(
       break;
   }
 
+  if (message_.is_atomic()) {
+    // Check the exists of memory scope and memory semantics ids.
+    auto memory_scope_instruction =
+        ir_context->get_def_use_mgr()->GetDef(message_.memory_scope_id());
+    auto memory_semantics_instruction =
+        ir_context->get_def_use_mgr()->GetDef(message_.memory_semantics_id());
+
+    if (!memory_scope_instruction || !memory_semantics_instruction) {
+      return false;
+    }
+    // The memory scope and memory semantics instructions must have the
+    // 'OpConstant' opcode.
+    if (memory_scope_instruction->opcode() != SpvOpConstant) {
+      return false;
+    }
+    if (memory_semantics_instruction->opcode() != SpvOpConstant) {
+      return false;
+    }
+
+    // The memory scope and memory semantics instructions must have an Integer
+    // operand type with signedness does not matters.
+    if (!ir_context->get_type_mgr()
+             ->GetType(memory_scope_instruction->type_id())
+             ->AsInteger()) {
+      return false;
+    }
+    if (!ir_context->get_type_mgr()
+             ->GetType(memory_semantics_instruction->type_id())
+             ->AsInteger()) {
+      return false;
+    }
```

Comment:
You still also need to check that the integer has 32 bits.

Reply:
Done.